### PR TITLE
Con 465 create page to emulate reserve funding

### PR DESCRIPTION
**Pull request by Ashley-Bass**

### CODE REVIEW

File: src/SFA.DAS.EmployerAccounts.Web/Views/EmployerTeam/V2/CheckFunding.cshtml [hunk 1]
```diff
@@ -1,7 +1,14 @@
-﻿
+﻿@using SFA.DAS.EmployerAccounts.Web.FeatureToggles
 @model SFA.DAS.EmployerAccounts.Web.ViewModels.AccountDashboardViewModel
 
 <h3 class="das-panel__heading">Check if government funding is available for apprenticeship training </h3>
 <p>Use government funding to pay for part or all of the cost of apprenticeship training.</p>
 
-<p><a href="#" class="das-panel__link">Check and secure funding now</a> </p>
+@if (Features.EmulatedFundingJourney.Enabled)
```
Comment (Stephek) [inline src/SFA.DAS.EmployerAccounts.Web/Views/EmployerTeam/V2/CheckFunding.cshtml]: Can this go into a URL helper (HtmlHelperExtensions extension methods).  Then the view doesn't know about the feature toggle.

File: src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerTeamController.cs [hunk 2]
```diff
@@ -352,5 +358,21 @@ public ActionResult CreateVacancy(AccountDashboardViewModel model)
         {
             return PartialView(model);
         }
+        [HttpGet]
+        [Route("EmulatedReserveFunding")]
```
Comment (Stephek) [inline src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerTeamController.cs]: Please move these methods to the EmulatedFundingJourneyController and update the link to point to them.

File: src/SFA.DAS.EmployerAccounts.Web/Views/EmployerTeam/V2/CheckFunding.cshtml (hunk 1 above)
Comment (Ashley-Bass) [reply src/SFA.DAS.EmployerAccounts.Web/Views/EmployerTeam/V2/CheckFunding.cshtml]: Done

File: src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerTeamController.cs (hunk 2 above)
Comment (Ashley-Bass) [reply src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerTeamController.cs]: Deleted the EmulatedReserveFunding method the others need to be here to allow for receiving of data

File: src/SFA.DAS.EmployerAccounts.Web/Views/EmployerTeam/V2/CheckFunding.cshtml (hunk 1 above)
Comment (Ashley-Bass) [reply src/SFA.DAS.EmployerAccounts.Web/Views/EmployerTeam/V2/CheckFunding.cshtml]: Prroperly done now

File: src/SFA.DAS.EmployerAccounts.Web/Controllers/EmulatedFundingJourneyController.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Route]
+        public ActionResult SendEvent(EmulatedFundingViewModel model)
+        {
+            if (model.PublishEvent)
+            {
+                _eventPublisher.Publish(new ReserveFundingCreatedEvent
+                {
+                    HashedAccountId = model.HashedAccountId,
```
Comment (lazcool) [inline src/SFA.DAS.EmployerAccounts.Web/Controllers/EmulatedFundingJourneyController.cs]: This is what the real event currently looks like. We should match it here.
```
    {
        "$type": "SFA.DAS.Reservations.Messages.ReservationCreatedEvent, SFA.DAS.Reservations.Messages",
        "Id": "02f6753f-78b9-4ac1-399e-a2167a7c986f",
        "AccountLegalEntityId": 776353,
        "AccountLegalEntityName": "Test Name",
        "CourseId": "1",
        "StartDate": "2019-08-02T00:00:00",
        "CourseName": "Network engineer",
        "EndDate": "2020-02-29T00:00:00",
        "CreatedDate": "2019-05-08T09:02:23.0693812Z"
    }
```

File: src/SFA.DAS.EmployerAccounts.Web/Views/EmulatedFundingJourney/Index.cshtml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        <span class="form-label-bold">Course Code</span>
+        <input class="form-control form-control-3-4" id="CourseCode" name="CourseCode" type="text" aria-required="true" value="@Model.CourseCode" />
+        <span class="form-label-bold">Start Date</span>
+        <input class="form-control form-control-3-4" id="StartDate" name="StartDate" type="text" aria-required="true" value="@Model.StartDate" />
+        <span class="form-label-bold">End Date</span>
+        <input class="form-control form-control-3-4" id="EndDate" name="EndDate" type="text" aria-required="true" value="@Model.EndDate" />
+        <span class="form-label-bold">Reservation ID</span>
+        <input class="form-control form-control-3-4" id="ReservationId" name="ReservationId" type="number" aria-required="true" value="@Model.ReservationId" placeholder=""/>
+        <span class="form-label-bold">Delay Event Publication</span>
+        <input class="form-control form-control-3-4" id="PublishEvent" name="PublishEvent" type="text" aria-required="true" value="@Model.PublishEvent" placeholder="True or False"/>
```
Comment (lazcool) [inline src/SFA.DAS.EmployerAccounts.Web/Views/EmulatedFundingJourney/Index.cshtml]: It would be nice if this was a `select`.

**Pull request closed without merging**